functions/imagemagick/main.py

```diff
@@ -0,0 +1,86 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# [START functions_imagemagick_setup]
+import os
+import subprocess
+
+from google.cloud import storage, vision
+
+
+storage_client = storage.Client()
+vision_client = vision.ImageAnnotatorClient()
+# [END functions_imagemagick_setup]
+
+
+# [START functions_imagemagick_analyze]
+# Blurs uploaded images that are flagged as Adult or Violence.
+def blur_offensive_images(data, context):
+    file = data
+
+    # Exit if this is a deletion or a deploy event.
+    if 'name' not in file:
+        print('This is a deploy event.')
+        return
+    elif file.get('resource_state', None) == 'not_exists':
+        print('This is a deletion event.')
+        return
+
+    file_name = file['name']
+    blob = storage_client.bucket(file['bucket']).get_blob(file_name)
+    blob_uri = 'gs://%s/%s' % (file['bucket'], file_name)
```

Comment:
Is there an attribute on the blob object itself that contains the full URI? If not, we can construct it like this.

Reply:
No - the blob object's attributes have the form `b/[BUCKET]/o/[OBJECT]`, whereas the Vision API requires `gs://[BUCKET]/[OBJECT]`.